packages/dataviews/src/types.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+type Item = Record< string, any >;
+
+interface Option {
+	value: any;
+	label: string;
+}
+
+interface filterByConfig {
+	operators?: string[];
+	isPrimary?: boolean;
+}
+
+export interface Field {
```

Comment:
I tried looking at all of our existing "fields" to define this. It might not be 100% accurate yet. We're also lacking the enumerations for available types and operators but it's a decent start.

Reply:
I'm not familiar with this package so this may be off base.

There's an alternative to defining a single interface that all the instances "fit," which is to make this a union of very specific types. If there's a specific set of known fields, a discriminated union can be more expressive and provide much more precise and relevant to developers working with the code.

```
// Instead of
interface Field {
  id: string;
  getValue?: ( obj: any ) => any;
}

// A union can be more expressive:
type FieldUnion =
  | { id: "countField"; getValue( field: { items: Array<any> } ) => number; }
  | { id: "dateField"; getValue( field: { date: Date } ) => string; };
```

Something to consider. [I also wrote a longer form post about the idea.](https://sirre.al/2023/09/06/these-types-are-not-the-same/)